optimum/habana/transformers/trainer.py

```diff
@@ -1023,21 +1028,12 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args, use_reentrant: Optio
                     context = (
                         functools.partial(self.accelerator.no_sync, model=model)
                         if i != len(batch_samples) - 1
-                        and self.accelerator.distributed_type != GaudiDistributedType.DEEPSPEED
+                        and self.accelerator.distributed_type != DistributedType.DEEPSPEED
                         else contextlib.nullcontext
                     )
                     with context():
                         tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
 
-                    if (
-                        args.parallel_mode == ParallelMode.DISTRIBUTED
-                        and args.distribution_strategy == "fast_ddp"
-                        and do_sync_step
-                    ):
-                        all_reduce_gradients(
```

Comment:
@IlyasMoutawwakil why did you remove the all_reduce_gradients?

Reply:
@regisss if we are removing fast ddp path we should update the document?

Reply:
"update the document", which document ?

Reply:
> "update the document", which document ?

https://github.com/huggingface/optimum-habana/blob/transformers_future/docs/source/tutorials/distributed.mdx
https://github.com/huggingface/optimum-habana/blob/transformers_future/docs/source/usage_guides/accelerate_training.mdx

Reply:
I remember the torch PATH was slow in some cases that is why all_reduce_gradients being added. We should check this

Reply:
Yes I updated the docs https://github.com/huggingface/optimum-habana/pull/1741/files#diff-c99090324b3019172c9ada4d3b9af205c0c5057fddbf4da9cf910a166ea55efe

Reply:
I am not sure about the impact of removing fast ddp. We have to check. I also need to check with @libinta